### PR TITLE
ColorPicker: Add accessible label for copy button

### PR DESCRIPTION
**Pull request by mirka**
Alternative to #57168
Addresses ##57157 (maybe not a full fix)

## What?

Adds an accessible label for the copy button in ColorPicker.

## Why?

The icon button was only associated with an ARIA description via the Tooltip, and didn't have an ARIA label. This is a pretty big bug for which the fix was stalled in #57168, so I'm proposing we address this now in a less controversial way.

### Testing Instructions for Keyboard

See the Storybook story for ColorPicker and check that the copy button is accessibly labeled.

### CODE REVIEW

File: packages/components/src/color-picker/color-copy-button.tsx [hunk 1]
```diff
@@ -65,6 +65,11 @@ export const ColorCopyButton = ( props: ColorCopyButtonProps ) => {
 		>
 			<CopyButton
 				size="small"
+				aria-label={
```
Comment (mirka) [inline packages/components/src/color-picker/color-copy-button.tsx]: This shouldn't result in a redundant ARIA description, thanks to #65989.

File: packages/components/src/color-picker/color-copy-button.tsx [hunk 2]
```diff
@@ -65,6 +65,11 @@ export const ColorCopyButton = ( props: ColorCopyButtonProps ) => {
...
+					copiedColor === color.toHex()
```
Comment (tyxla) [inline packages/components/src/color-picker/color-copy-button.tsx]: Could we put this in a variable? This is literally the same we're passing over to the `text` prop, so we could avoid the duplication.

**Review by tyxla — APPROVED** [review]
LGTM 👍 

Just one tiny deduplication suggestion before 🚢

File: packages/components/src/color-picker/color-copy-button.tsx (hunk 2 above)
Comment (mirka) [reply packages/components/src/color-picker/color-copy-button.tsx]: Good point!

**Pull request merged**